hw/drivers/chg_ctrl/da1469x_charger/src/da1469x_charger.c

```diff
@@ -0,0 +1,456 @@
+    case CHARGER_STATE_TDIE_PROT:
+    case CHARGER_STATE_TBAT_PROT:
+        *fault |= CHARGE_CONTROL_FAULT_THERM;
+        break;
+    default:
+        break;
+    }
+
+    if (CHARGER->CHARGER_STATUS_REG &
+        CHARGER_CHARGER_STATUS_REG_VBAT_OVP_COMP_OUT_Msk) {
+        *fault |= CHARGE_CONTROL_FAULT_OV;
+    }
+
+    return 0;
+}
+
+static int
+da1469x_charger_chg_ctrl_enable(struct charge_control *cc)
+{
+    return da1469x_charger_charge_enable((struct da1469x_charger_dev *)cc->cc_dev);
+}
+
+static int
+da1469x_charger_chg_ctrl_disable(struct charge_control *cc)
+{
+    return da1469x_charger_charge_disable((struct da1469x_charger_dev *)cc->cc_dev);
+}
+
+/* Exports for the charge control API */
+static const struct charge_control_driver da1469x_charger_chg_ctrl_driver = {
+    .ccd_read = da1469x_charger_chg_ctrl_read,
+    .ccd_get_config = da1469x_charger_chg_ctrl_get_config,
+    .ccd_set_config = da1469x_charger_chg_ctrl_set_config,
+    .ccd_get_status = da1469x_charger_chg_ctrl_get_status,
+    .ccd_get_fault = da1469x_charger_chg_ctrl_get_fault,
+    .ccd_enable = da1469x_charger_chg_ctrl_enable,
+    .ccd_disable = da1469x_charger_chg_ctrl_disable,
+};
+#endif /* DA1469X_CHARGER_USE_CHARGE_CONTROL */
+
+int
+da1469x_charger_init(struct os_dev *dev, void *arg)
+{
+    struct da1469x_charger_dev *charger = (struct da1469x_charger_dev *)dev;
+#if MYNEWT_VAL(DA1469X_CHARGER_USE_CHARGE_CONTROL)
+    struct charge_control *cc;
+#endif
+    const struct da1469x_charger_config *cfg = arg;
+    int rc = 0;
+
+    if (!dev) {
+        rc = SYS_ENODEV;
+        goto err;
+    }
+
+    if (cfg) {
+        rc = da1469x_charger_set_config(charger, cfg);
+        if (rc) {
+            goto err;
+        }
+    }
+
+#if MYNEWT_VAL(DA1469X_CHARGER_USE_CHARGE_CONTROL)
+    cc = &charger->chg_ctrl;
+
+    rc = charge_control_init(cc, dev);
+    if (rc) {
+        goto err;
+    }
+
+    /* Add the driver with all the supported types */
+    rc = charge_control_set_driver(cc, CHARGE_CONTROL_TYPE_STATUS |
+        CHARGE_CONTROL_TYPE_FAULT,
+        (struct charge_control_driver *)&da1469x_charger_chg_ctrl_driver);
+    if (rc) {
+        goto err;
+    }
+
+    charge_control_set_type_mask(cc,
+                                 CHARGE_CONTROL_TYPE_STATUS |
+                                 CHARGE_CONTROL_TYPE_FAULT);
+
+    rc = charge_control_mgr_register(cc);
+    if (rc) {
+        goto err;
+    }
+#endif /* DA1469X_CHARGER_USE_CHARGE_CONTROL */
+
+    da1469x_charger_state_changed_event.ev_arg = charger;
+    da1469x_charger_error_event.ev_arg = charger;
+
+    da1469x_charger_set_error_irq_mask(charger, CHARGER_ERROR_IRQ_NONE);
+    da1469x_charger_set_state_change_irq_mask(charger, CHARGER_STATE_IRQ_NONE);
+
+    NVIC_SetVector(CHARGER_STATE_IRQn, (uint32_t)CHARGER_STATE_IRQHandler);
+    NVIC_SetPriority(CHARGER_STATE_IRQn, (1U << __NVIC_PRIO_BITS) - 1);
```

Comment:
I think these calls to NVIC_SetPriority() are unnecessary. OS init already sets the priority to be as low as it can be.